classes/controller/WSController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+use PrestaShop\PrestaShop\Adapter\ContainerBuilder;
+
+class WSControllerCore extends Controller
+{
+    /**
+     * WSControllerCore constructor.
+     *
+     * @throws PrestaShopException
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->controller_type = 'ws';
+
+        $this->init();
+    }
+
+    /**
+     * @return bool|void
+     *
+     * @throws Exception
+     */
+    public function checkAccess()
+    {
+        throw new Exception('Method checkAccess() not implemented in WSController');
```

Comment:
So an exception is always raised when the controller is running :thinking: 
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/controller/Controller.php#L273

Reply:
We maybe need to create  a default value sometimes, depending on webservice context.

Reply:
No, because in our case, we construct and init the controller but there is no run.